components/expression/expression.js

```diff
@@ -30,6 +30,70 @@ genotet.ExpressionView = function(viewName, params) {
   $(this.container).on('genotet.ready', function() {
     this.loader.load(params.matrixName, params.geneRegex, params.condRegex);
   }.bind(this));
+
+  // Set up rendering update.
+  $(this.panel)
+    .on('genotet.update', function(event, data) {
+      switch (data.type) {
+        case 'label':
+          this.renderer.render();
+          break;
+        case 'visibility':
+          this.renderer.render();
+          break;
+        case 'gene':
+          this.loader.updateGenes(data.method, data.regex, params);
+          break;
+        case 'condition':
+          this.loader.updateConditions(data.method, data.regex, params);
+          break;
+        case 'auto-scale':
+          this.renderer.render(this.data.options.autoScaleGradient);
+          break;
+        default:
+          genotet.error('unknown update type', data.type);
+      }
+    }.bind(this))
+    .on('genotet.addGeneProfile', function(event, geneIndex) {
+      this.renderer.addGeneProfile_(geneIndex);
+    }.bind(this))
+    .on('genotet.removeGeneProfile', function(event, geneIndex) {
+      this.renderer.removeGeneProfile_(geneIndex);
+    }.bind(this));
+
+  // Cell hover in expression.
+  $(this.renderer)
+    .on('genotet.cellHover', function(event, cell) {
+      this.renderer.highlightHoverCell_(cell, true);
```

Comment:
It is recommended to sperate highlight and unhighlight into two functions, rather than passing a boolean parameter.